pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -90,7 +90,7 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 		return distribution.Descriptor{}, err
 	}
 	pullthroughBlobStore := repo.Blobs(ctx)
-	desc, err := pullthroughBlobStore.Stat(r.repo.ctx, dgst)
+	desc, err := pullthroughBlobStore.Stat(ctx, dgst)
```

Comment:
I'm sorry again. I was looking at the wrong `Stat()`. The [`Stat()`](https://github.com/miminar/origin/blob/registry-quota-refactor/Godeps/_workspace/src/github.com/docker/distribution/registry/client/repository.go#L493) is invoked on the registry client created [here](https://github.com/miminar/origin/blob/registry-quota-refactor/pkg/image/importer/importer.go#L642). So the `unauthorized` really comes from the `docker.io`